components/layout/display_list/stacking_context.rs

```diff
@@ -943,7 +944,7 @@ impl BoxFragment {
             return Some(StackingContextType::FloatStackingContainer);
         }
 
-        if self.is_atomic_inline_level() {
+        if self.is_atomic_inline_level() || flags.contains(FragmentFlags::IS_FLEX_OR_GRID_ITEM) {
```

Comment:
Maybe put a link here to where this is described in the specification before landing?